test/AutoDiff/Sema/DerivedConformances/enum_additive_arithmetic.swift

```diff
@@ -0,0 +1,64 @@
+// RUN: %target-swift-frontend -typecheck -verify -primary-file %s
```

Comment:
Note if you pick up this work later: runtime tests should also be added to verify derived conformances behavior.

Example ([reference test](https://github.com/apple/swift/blob/3e090b483352b3c6f889eec521605fda3ec1ab51/test/AutoDiff/validation-test/simple_math.swift)):
```swift
// RUN: %target-run-simple-swift
// REQUIRES: executable_test

import StdlibUnittest
import DifferentiationUnittest

var EnumSynthesisTests = TestSuite("EnumAdditiveArithmeticSynthesis")

enum Ints: AdditiveArithmetic {
  case branch(Int, Int)
  case leaf(Int)
  case none
}

EnumSynthesisTests.test("Ints") {
  let ints1 = Ints.branch(1, 2)
  let ints2 = Ints.branch(2, 1)
  expectEqual(.branch(3, 3), ints1 + ints2)
}

runAllTests()
```

`test/AutoDiff/Sema/DerivedConformances/enum_additive_arithmetic_runtime.swift` is a good location.